### PR TITLE
[cross-entropy-loss] Added support for DFT flag

### PR DESCRIPTION
**Pull request by kashif**
## Summary
Added support for a flag that turns on the DFT cross entropy loss from the paper https://arxiv.org/abs/2508.05629

- Hardware Type: cuda
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Review by lancerts — APPROVED** [review]

**Comment by kashif** [comment]
@lancerts  sorry i have small fix for this PR in #864
